numpy/core/_methods.py

```diff
@@ -115,7 +115,7 @@ def _var(a, axis=None, dtype=None, out=None, ddof=0, keepdims=False):
     # Note that x may not be inexact and that we need it to be an array,
     # not a scalar.
     x = asanyarray(arr - arrmean)
-    if issubclass(arr.dtype.type, nt.complexfloating):
+    if not issubclass(arr.dtype.type, nt.integer):
```

Comment:
This should include also `nt.floating`. And might as well swap the branches, i.e.,
```
if issubclass(arr.dtype.type, (nt.floating, nt.integer)):
    x = um.multiply(x, x, out=x)
else:
    x = um.multiply(x, um.conjugate(x), out=x).real
```